internal/cli/cmd/connectivity.go

```diff
@@ -76,11 +77,17 @@ func newCmdConnectivityTest(hooks Hooks) *cobra.Command {
 				return err
 			}
 
-			ctx, _ := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
+			ctx, _ := signal.NotifyContext(cmd.Context(), os.Interrupt, syscall.SIGTERM)
+
+			if params.Timeout > 0 {
+				timeoutCtx, cancelCtx := context.WithTimeoutCause(ctx, params.Timeout, fmt.Errorf("connectivity test suite timeout (%s) reached", params.Timeout))
```

Comment:
nit: Suffix confusing. cancelCtx is not a context but a CancelFunc.

Reply:
renamed - thanks @derailed